crates/worker/src/main.rs

```diff
@@ -91,7 +98,27 @@ pub async fn run_workers(options: Options) {
         .unwrap();
 
     let job_port = Arc::new(JobDBRepository::new(pool.clone()));
-    let state = State::new(job_port);
+
+    let worker_adapter: Arc<dyn WorkerPort + Send + Sync> = Arc::new(
+        WorkerAdapter::new(
+            &options.pg.url,
+            options.pg.max_size,
+            options.worker.schema.clone(),
+        )
+        .await,
+    );
+
+    let app_config = AppConfig::from(EnvConfig::new());
+
+    let sequence_number = app_config
+        .client
+        .get_account(app_config.account.address())
+        .await
+        .unwrap()
+        .into_inner()
+        .sequence_number;
+    app_config.account.set_sequence_number(sequence_number);
+    let state = State::new(job_port, worker_adapter, app_config);
```

Comment:
config should already inside each port, adapter,...,
```suggestion
    let state = State::new(job_port, worker_adapter);
```